### PR TITLE
Use minimal summary listener for Suite Engine

### LINKED ISSUE #5710: Suite Engine logs "No TestIdentifier with unique ID" for failing tests
Since upgrading from junit platform 6.0.3 to 6.1.0, now getting below output in console log.

Junit 6.1.0
Cucumber 7.34.3
Maven surefire 3.5.5
```
WARNING: EngineExecutionListener [org.junit.platform.launcher.core.ExecutionListenerAdapter] threw exception for method: executionFinished(PickleDescriptor: [engine:junit-platform-suite]/[suite:runners.TestRunner]/[engine:cucumber]/[feature:<file>, TestExecutionResult [status = FAILED, throwable = java.lang.RuntimeException: <exception>)
org.junit.platform.commons.PreconditionViolationException: No TestIdentifier with unique ID [[engine:junit-platform-suite]/[suite:runners.TestRunner]] has been added to this TestPlan.
	at org.junit.platform.commons.util.Preconditions.condition(Preconditions.java:389)
	at org.junit.platform.commons.util.Preconditions.notNull(Preconditions.java:74)
	at org.junit.platform.launcher.TestPlan.getTestIdentifier(TestPlan.java:227)
	at java.base/java.util.Optional.map(Optional.java:260)
	at org.junit.platform.launcher.TestPlan.getParent(TestPlan.java:170)
	at org.junit.platform.launcher.listeners.MutableTestExecutionSummary.collectTestDescription(MutableTestExecutionSummary.java:235)
	at org.junit.platform.launcher.listeners.MutableTestExecutionSummary.lambda$collectTestDescription$0(MutableTestExecutionSummary.java:235)
	at java.base/java.util.Optional.ifPresent(Optional.java:178)
	at org.junit.platform.launcher.listeners.MutableTestExecutionSummary.collectTestDescription(MutableTestExecutionSummary.java:235)
	at org.junit.platform.launcher.listeners.MutableTestExecutionSummary.lambda$collectTestDescription$0(MutableTestExecutionSummary.java:235)
	at java.base/java.util.Optional.ifPresent(Optional.java:178)
	at org.junit.platform.launcher.listeners.MutableTestExecutionSummary.collectTestDescription(MutableTestExecutionSummary.java:235)
	at org.junit.platform.launcher.listeners.MutableTestExecutionSummary.lambda$collectTestDescription$0(MutableTestExecutionSummary.java:235)
	at java.base/java.util.Optional.ifPresent(Optional.java:178)
	at org.junit.platform.launcher.listeners.MutableTestExecutionSummary.collectTestDescription(MutableTestExecutionSummary.java:235)
	at org.junit.platform.launcher.listeners.MutableTestExecutionSummary.lambda$collectTestDescription$0(MutableTestExecutionSummary.java:235)
	at java.base/java.util.Optional.ifPresent(Optional.java:178)
	at org.junit.platform.launcher.listeners.MutableTestExecutionSummary.collectTestDescription(MutableTestExecutionSummary.java:235)
	at org.junit.platform.launcher.listeners.MutableTestExecutionSummary.describeTest(MutableTestExecutionSummary.java:229)
	at org.junit.platform.launcher.listeners.MutableTestExecutionSummary.addFailure(MutableTestExecutionSummary.java:80)
	at org.junit.platform.launcher.listeners.SummaryGeneratingListener.lambda$executionFinished$0(SummaryGeneratingListener.java:140)
	at java.base/java.util.Optional.ifPresent(Optional.java:178)
	at org.junit.platform.launcher.listeners.SummaryGeneratingListener.executionFinished(SummaryGeneratingListener.java:139)
	at org.junit.platform.launcher.core.ExecutionListenerAdapter.executionFinished(ExecutionListenerAdapter.java:57)
	at org.junit.platform.launcher.core.CompositeEngineExecutionListener.lambda$executionFinished$0(CompositeEngineExecutionListener.java:60)
	at org.junit.platform.launcher.core.CompositeEngineExecutionListener.lambda$notifyEach$0(CompositeEngineExecutionListener.java:82)
	at org.junit.platform.commons.util.CollectionUtils.forEachInReverseOrder(CollectionUtils.java:213)
	at org.junit.platform.launcher.core.IterationOrder$2.forEach(IterationOrder.java:30)
	at org.junit.platform.launcher.core.CompositeEngineExecutionListener.notifyEach(CompositeEngineExecutionListener.java:80)
	at org.junit.platform.launcher.core.CompositeEngineExecutionListener.executionFinished(CompositeEngineExecutionListener.java:59)
	at org.junit.platform.launcher.core.DelegatingEngineExecutionListener.executionFinished(DelegatingEngineExecutionListener.java:47)
	at org.junit.platform.launcher.core.StackTracePruningEngineExecutionListener.executionFinished(StackTracePruningEngineExecutionListener.java:46)
	at org.junit.platform.launcher.core.DelegatingEngineExecutionListener.executionFinished(DelegatingEngineExecutionListener.java:47)
	at org.junit.platform.launcher.core.OutcomeDelayingEngineExecutionListener.executionFinished(OutcomeDelayingEngineExecutionListener.java:71)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.reportCompletion(NodeTestTask.java:224)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:121)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:42)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$2(NodeTestTask.java:180)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:74)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$1(NodeTestTask.java:166)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:139)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$0(NodeTestTask.java:164)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:74)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:163)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:116)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:42)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$2(NodeTestTask.java:180)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:74)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$1(NodeTestTask.java:166)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:139)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$0(NodeTestTask.java:164)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:74)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:163)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:116)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:42)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$2(NodeTestTask.java:180)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:74)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$1(NodeTestTask.java:166)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:139)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$0(NodeTestTask.java:164)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:74)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:163)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:116)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:42)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$2(NodeTestTask.java:180)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:74)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$1(NodeTestTask.java:166)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:139)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$0(NodeTestTask.java:164)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:74)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:163)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:116)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:36)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:52)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:58)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.executeEngine(EngineExecutionOrchestrator.java:256)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.failOrExecuteEngine(EngineExecutionOrchestrator.java:228)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:189)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:109)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:90)
	at org.junit.platform.suite.engine.SuiteLauncher.execute(SuiteLauncher.java:65)
	at org.junit.platform.suite.engine.SuiteTestDescriptor.executeTests(SuiteTestDescriptor.java:208)
	at org.junit.platform.suite.engine.SuiteTestDescriptor.execute(SuiteTestDescriptor.java:173)
	at org.junit.platform.suite.engine.SuiteTestEngine.execute(SuiteTestEngine.java:80)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.executeEngine(EngineExecutionOrchestrator.java:256)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.failOrExecuteEngine(EngineExecutionOrchestrator.java:228)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:189)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:109)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:66)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:168)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:65)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:137)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:126)
	at org.junit.platform.launcher.core.DelegatingLauncher.execute(DelegatingLauncher.java:71)
	at org.junit.platform.launcher.core.InterceptingLauncher.lambda$execute$2(InterceptingLauncher.java:66)
	at org.junit.platform.launcher.core.ClasspathAlignmentCheckingLauncherInterceptor.intercept(ClasspathAlignmentCheckingLauncherInterceptor.java:25)
	at org.junit.platform.launcher.core.InterceptingLauncher.execute(InterceptingLauncher.java:65)
	at org.junit.platform.launcher.core.DelegatingLauncher.execute(DelegatingLauncher.java:71)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.apache.maven.surefire.api.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:125)
	at org.apache.maven.surefire.junitplatform.LauncherAdapter.executeWithCancellationToken(LauncherAdapter.java:68)
	at org.apache.maven.surefire.junitplatform.LauncherAdapter.execute(LauncherAdapter.java:54)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:203)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:168)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:136)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)
```

### PR DESCRIPTION
**Pull request by mpkorstanje**
The Suite engine uses the `SummaryGeneratingListener` to track if
any tests were discovered. After #5588 this fails because it now tries
to construct a complete description of the failing test. This assumes
that all ancestors of a test descriptor are in the plan. Which isn't
true for plans from the SuiteEngine, as these are a subtree of the
overall plan.

Using a minimal implementation of the `SummaryGeneratingListener` we
can avoid this problem.

Fixes: #5710

---

I hereby agree to the terms of the [JUnit Contributor License Agreement](https://github.com/junit-team/junit-framework/blob/002a0052926ddee57cf90580fa49bc37e5a72427/CONTRIBUTING.md#junit-contributor-license-agreement).

---

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Method [preconditions](https://docs.junit.org/snapshot/api/org.junit.platform.commons/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [x] [Coding conventions](https://github.com/junit-team/junit-framework/blob/HEAD/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [x] Change is covered by [automated tests](https://github.com/junit-team/junit-framework/blob/HEAD/CONTRIBUTING.md#tests) including corner cases, errors, and exception handling
- [x] Public API has [Javadoc](https://github.com/junit-team/junit-framework/blob/HEAD/CONTRIBUTING.md#javadoc) and [`@API` annotations](https://apiguardian-team.github.io/apiguardian/docs/current/api/org/apiguardian/api/API.html)
- [x] Change is documented in the [User Guide](https://docs.junit.org/snapshot/) and [Release Notes](https://docs.junit.org/snapshot/release-notes.html)

### CODE REVIEW

File: junit-platform-suite-engine/src/main/java/org/junit/platform/suite/engine/SuiteSummaryGeneratingListener.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * test descriptors are in the test plan. This isn't true for the suite engine
+ * which only executes a subtree of the test plan. This implementation tracks
+ * only the essentials.
+ */
+final class SuiteSummaryGeneratingListener implements TestExecutionListener {
+	private final AtomicLong testsFound = new AtomicLong();
+
+	@Override
+	public void testPlanExecutionStarted(TestPlan testPlan) {
+		this.testsFound.set(testPlan.countTestIdentifiers(TestIdentifier::isTest));
```
Comment (mpkorstanje) [inline junit-platform-suite-engine/src/main/java/org/junit/platform/suite/engine/SuiteSummaryGeneratingListener.java]: Not happy with this yet. This is ultimately derived from the `LauncherDiscoveryResult` which we can access earlier. So this listener could be eliminated entirely.

Comment (mpkorstanje) [reply junit-platform-suite-engine/src/main/java/org/junit/platform/suite/engine/SuiteSummaryGeneratingListener.java]: Using the listener however matches the way the console launcher implements `--fail-if-no-tests` which keeps things consistent.

File: documentation/modules/ROOT/partials/release-notes/release-notes-6.2.0-M1.adoc [hunk 2]
```diff
@@ -16,7 +16,7 @@ repository on GitHub.
 [[v6.2.0-M1-junit-platform-bug-fixes]]
 ==== Bug Fixes
 
-* ❓
+* The Suite Engine no longer logs "No TestIdentifier with unique ID" for failing tests.
```
Comment (marcphilipp) [inline documentation/modules/ROOT/partials/release-notes/release-notes-6.2.0-M1.adoc]: This should go to the 6.1.1 release notes file which I just created on `main`.

**Review by marcphilipp — APPROVED** [review]
Looks good to me except for release notes comment. Thanks! 👍
